sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/InMemoryCatalog.scala

```diff
@@ -277,9 +277,18 @@ class InMemoryCatalog extends ExternalCatalog {
     catalog(db).tables(table).partitions(spec)
   }
 
+  /**
+   * List the metadata of all partitions that belong to the specified table, assuming it exists.
+   *
+   * A partial partition spec may optionally be provided to filter the partitions returned.
+   * For instance, if there exist partitions (a='1', b='2'), (a='1', b='3') and (a='2', b='4'),
+   * then a partial spec of (a='1') will return the first two only.
+   * TODO: Currently partialSpec is not used for memory catalog and it returns all the partitions.
```

Comment:
Can we throw exception instead of returning all the partitions if a partition spec is given to an in-memory catalog? Silently returning wrong answer can be dangerous.